doc/specs/#5000 - Process Model 2.0/#4472 - Windows Terminal Session Management.md

```diff
@@ -0,0 +1,528 @@
+user was typing in, the current one. However, if someone were to do something
+like `sleep 10 ; wt -w 0 <commands>`, then the user could easily focus another
+WT window during the sleep, which would cause the MRU window to not be the same
+as the window executing the command.
+
+I'm not sure that there is a better solution for the `-w 0` scenario other than
+attempting to use the `WT_SESSION` environment variable. If a `wt.exe` process
+is spawned and that's in it's environment variables, it could try and ask the
+monarch for the peasant who's hosting the session corresponding to that GUID.
+This is more of a theoretical solution than anything else.
+
+In Single-Instance mode, running `wt -w 0` outside a WT window will still cause
+the commandline to glom to the existing single terminal instance, if there is
+one.
+
+#### Running commands in a new window:`wt --window -1` / `wt --window new`
+
+If the user passes a negative number, or the reserved name `new` to the
+`--window` parameter, then we will always create a new window for that
+commandline, regardless of the value of `glomToLastWindow`. This will allow
+users to do something like `wt -w -1 new-tab` to _always_ create a new window.
+
+#### `--window` in subcommands
+
+The `--window` parameter is a setting to `wt.exe` itself, not to one of its
+subcommands (like `new-tab` or `split-pane`). This means that all of the
+subcommands in a particular `wt` commandline will all be handled by the same
+session. For example, let us consider a user who wants to open a new tab in
+window 2, and split a new pane in window 3, all at once. The user _cannot_ do
+something like:
+
+```cmd
+wt -w 2 new-tab ; -w 3 split-pane
+```
+
+Instead, the user will need to separate the commands (by whatever their shell's
+own command delimiter is) and run two different `wt.exe` instances:
+
+```cmd
+wt -w 2 new-tab & wt -w 3 split-pane
+```
+
+This is done to make the parsing of the subcommands easier, and for the internal
+passing of arguments simpler. If the `--window` parameter were a part of each
+subcommand, then each individual subcommand's parser would need to be
+enlightened about that parameter, and then it would need to be possible for any
+single part of the commandline to call out to another process. It would be
+especially tricky then to coordinate the work being done across process here.
+The source process would need some sort of way to wait for the other process to
+notify the source that a particular subcommand completed, before allowing the
+source to dispatch the next part of the commandline.
+
+Overall, this is seen as unnecessarily complex, and dispatching whole sets of
+commands as a simpler solution.
+
+### Naming Windows
+
+It's not user-friendly to rely on automatically generated, invisible numbers to
+identify windows. There's not a great way of identifying which window is which.
+The user would need to track the IDs in their head manually. Instead, we'll

```

Comment:
I'm thinking we just reject attempts to name a window the same thing as an existing window. I'm working on a "toast" that we can use to display a temporary, low-impact message to the user. That'd probably be what we use to display the message, rather than a high-impact `Dialog`. Something like "Unable to rename window. Another window with that name already exists."